src/js/background/assignManager.js

```diff
@@ -294,14 +323,51 @@ const assignManager = {
     // We also can't change for always private mode
     // See: https://bugzilla.mozilla.org/show_bug.cgi?id=1352102
     browser.contextMenus.remove(this.MENU_ASSIGN_ID);
+    browser.contextualIdentities.query({}).then((identities) => {
+      identities.forEach((identity) => {
+        this.removeContainerMenuEntry(identity);
+      });
+    }).catch(() => {});
+    browser.contextMenus.remove(this.MENU_REOPEN_IN);
     browser.contextMenus.remove(this.MENU_REMOVE_ID);
     browser.contextMenus.remove(this.MENU_SEPARATOR_ID);
     browser.contextMenus.remove(this.MENU_HIDE_ID);
     browser.contextMenus.remove(this.MENU_MOVE_ID);
   },
 
+  addContainerMenuEntry(contextualIdentity, contexts) {
+    browser.contextMenus.create({
+      id: this.cookieStoreId2menuId(contextualIdentity.cookieStoreId),
+      title: contextualIdentity.name,
+      // TODO: colorized icons?
+      icons: {
+        "16": contextualIdentity.iconUrl
+      },
+      // TODO: hide entry for current container in context menu of tabs
+      contexts: contexts,
```

Comment:
Can the calling function just not call this function when the context id matches?